tests/test_transforms/test_decomposition/test_stl_transform.py

```diff
@@ -233,3 +241,9 @@ def test_params_to_tune(ts_trend_seasonal):
     transform = STLTransform(in_column="target", period=7)
     assert len(transform.params_to_tune()) > 0
     assert_sampling_is_valid(transform=transform, ts=ts)
+
+
+@pytest.mark.parametrize("model_stl", [10, custom_model])
+def test_custom_model(model_stl):
+    with pytest.raises(ValueError, match="Model should be a string or TimeSeriesModel"):
+        transform = STLTransform(in_column="target", period=7, model=model_stl)
```

Comment:
Add test like in issue with ETSModel

Reply:
This test checks the case like in issue and an additional case, if, for example, `type(model) == int`